solo/cli/key.py

```diff
@@ -133,11 +133,13 @@ def make_credential(serial, host, user, udp, prompt, pin):
 
     import solo.hmac_secret
 
-    #check for PIN
-    if not pin:
-        pin = getpass.getpass("PIN (leave empty for no PIN: ")
-    if not pin:
+    dev = solo.client.find(serial, udp=udp)
+    if dev.has_pin() is False:
+        #remove PIN if no PIN set
```

Comment:
silently removing the PIN is incorrect and misleading -- if there is no PIN set and the user specifies a PIN, he should get an error

Reply:
okay yeah silently is maybe bad, I just wanted to try and resolve this without the user needing to actially do anything extra, making it easier to use